course_catalog/etl/video_test.py

```diff
@@ -0,0 +1,126 @@
+                },
+            }
+        ],
+        "kind": "youtube#searchListResponse",
+        "pageInfo": {"resultsPerPage": 50, "totalResults": 51},
+        "regionCode": "US",
+    }
+
+    if token != "":
+        data["nextPageToken"] = token
+
+    return data
+
+
+def mock_video_data():
+    """Mock video data"""
+    data = {
+        "kind": "youtube#videoListResponse",
+        "etag": '"j6xRRd8dTPVVptg711_CSPADRfg/j0RF7LBD1f8ViAQN-D94fJ40WZ8"',
+        "pageInfo": {"totalResults": 1, "resultsPerPage": 1},
+        "items": [
+            {
+                "kind": "youtube#video",
+                "etag": '"j6xRRd8dTPVVptg711_CSPADRfg/iuXInUCcGHCzpxuwYfW_Of_a7IA"',
+                "id": "1jrHVNJ4Zds",
+                "snippet": {
+                    "publishedAt": "2018-08-14T18:40:28.000Z",
+                    "channelId": "UCTBMWu8yshnAmpzR3MoJFtw",
+                    "title": "Making Biologic Medicines for Patients: The Bioreactor Deconstructed",
+                    "description": "description 1",
+                    "channelTitle": "MITx Videos",
+                    "categoryId": "22",
+                    "liveBroadcastContent": "none",
+                    "localized": {
+                        "title": "Making Biologic Medicines for Patients: The Bioreactor Deconstructed",
+                        "description": "description 1",
+                    },
+                    "defaultAudioLanguage": "en-US",
+                },
+            }
+        ],
+    }
+
+    return data
+
+
+@pytest.fixture
+def mocked_channel_data():
+    """Mock video channel yml"""
+    data = """
+---
+channels:
+  - offered_by: MIT
+    channel_id: channel1
+  - offered_by: OCW
+    channel_id: channel2
+"""
+
+    return data
+
+
+@pytest.fixture
+def mocked_channel_response(mocked_responses, mocked_channel_data, settings):
+    """Mock the catalog response"""
+    mocked_responses.add(
+        mocked_responses.GET,
+        settings.VIDEOS_CONFIG_FILE_LOCATION,
+        body=mocked_channel_data,
+    )
+    yield mocked_responses
+
+
+@pytest.mark.usefixtures("mocked_channel_response")
+def test_extract(mocker):
+    """Test video ETL extract"""
+    mock_video_client = mocker.patch("course_catalog.etl.video.get_youtube_client")
+    mock_video_client.return_value.search.return_value.list.return_value.execute.side_effect = [
+        mock_video_search_data("token"),
+        mock_video_search_data(""),
+        mock_video_search_data(""),
+    ]
+
+    mock_video_client.return_value.videos.return_value.list.return_value.execute.return_value = (
+        mock_video_data()
+    )
+
+    generator = video.extract()
+
+    result1 = next(generator)
+    assert result1 == ("MIT", mock_video_data())
+
+    result2 = next(generator)
+    assert result2 == ("MIT", mock_video_data())
+
+    result3 = next(generator)
+    assert result3 == ("OCW", mock_video_data())
+
+    with pytest.raises(StopIteration):
+        next(generator)
```

Comment:
A cleaner way to write this would be:

```python
assert list(video.extract()) == [
    ("MIT", mock_video_data()),
    ("MIT", mock_video_data()),
    ("OCW", mock_video_data())
]
```
```